packages/dcos-integration-test/extra/conftest.py

```diff
@@ -39,13 +40,47 @@ def pytest_addoption(parser):
                      help="run only Windows tests")
 
 
+def _add_xfail_markers(item):
```

Comment:
This is nice! but it is certainly adding to the runtime cost with parsing and processing the marker everytime pytest encounters it.

Reply:
Is there a better way to introduce the `date=` field? FWIW, I was not not very opposed `common.py` and I foresaw it being useful for more common helper functions.

Reply:
@orsenthil Have a look at the description for how one could do this in `common.py`.

Reply:
> This is nice! but it is certainly adding to the runtime cost with parsing and processing the marker everytime pytest encounters it.

Yes but that is tiny I believe compared to the HTTP requests made by the tests.

Reply:
>  I foresaw it being useful for more common helper functions.

Why would such functions not exist in `dcos-test-utils`?

Reply:
> Why would such functions not exist in dcos-test-utils?

In-repo vs outside repo dependency (which brings in a need to bump, test, iterate).

Reply:
> @orsenthil Have a look at the description for how one could do this in common.py.

ACK. TBH, I would have gone that way, the way you demonstrated in the description. 

But, this looks good to me as well.

Reply:
> In-repo vs outside repo dependency (which brings in a need to bump, test, iterate).
 
Yes, this is the problem with Test Utils, but hopefully we can make that easier!

@orsenthil Thank you - I'm happy to continue iterating! The main thing here is the keywords.

Reply:
This LGTM! 

Please let me know when you are done, this is an excellent improvement, hopefully it still keeps the dashboard sane.